Qspi hal test refactoring/update

maciejbocianski opened the pull request:
### Description

- code refactoring and preparation for enabling DPI/QPI tests
- add DPI and QPI support
- reduced multiple test count to 4
- use common flash config header for all MX25RXX35F chips
- fix sector erase max time on N25Q128A
- randomize flash address during test to extend flash life (CI boards)



### Pull request type
    [ ] Fix
    [X] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Breaking change

maciejbocianski committed:
qspi_hal_test refactoring

- code refactoring and preparation for enabling DPI/QPI tests
- reduced multiple test count to 4
- use common flash config header for all MX25RXX35F chips
- fix sector erase max time on N25Q128A

maciejbocianski committed:
qspi_hal_test add DPI and QPI support

maciejbocianski committed:
qspi_hal_test - randomize flash RW addres

randomize flash address during test to extend flash life

maciejbocianski commented:
@jamesbeyond @0xc0170 @SenRamakri @offirko 
please review it

offirko reviewed (approved):
Looks good to me

maciejbocianski committed:
qspi_hal_test - remove unused variables

maciejbocianski commented:
Tested on:
```
DISCO_F413ZH 
DISCO_L475VG_IOT01A
EFM32GG11_STK3701
NRF52840_DK 
```

0xc0170 reviewed (approved):
test utils look like they might become driver update or keep driver as it is ?

maciejbocianski committed:
fix qspi address sending for nrf52

fix address sending in qspi_command_transfer
now address is send MSB first

maciejbocianski commented:
~~Don't know why, but on nrf52840_dk only first 16 sectors of MX25R6435F are writable (checked: no protection was enabled). 
Patch for test: use only first 16 blocks during test https://github.com/ARMmbed/mbed-os/pull/7925/commits/def7ee10a3da37a9c9c6a157254bbe070322f8cd~~

@0xc0170 
There was a bug in [qspi_command_transfer](https://github.com/ARMmbed/mbed-os/blob/2e869df2969ee0d414091694b057668812ead2e2/targets/TARGET_NORDIC/TARGET_NRF5x/qspi_api.c#L302) function, address was send LSB first but should be MSB first
Fix for this: https://github.com/ARMmbed/mbed-os/pull/7925/commits/3a64383de57ae42faf3b76ecfc096f6771a54b44

maciejbocianski committed:
qspi_hal_test - adjust used memory sector count

set flash sector count to 1024 for EFM32GG11_STK3701(MX25R3235F)
set flash sector count to 2048 for NRF52840_DK(MX25R6435F)

maciejbocianski commented:
@0xc0170 @jamesbeyond @offirko 
please review again, two commits were added
https://github.com/ARMmbed/mbed-os/pull/7925/commits/3a64383de57ae42faf3b76ecfc096f6771a54b44
https://github.com/ARMmbed/mbed-os/pull/7925/commits/cffd581308c3783cc32ae3955cb4770f30a75534

jamesbeyond reviewed (approved):
Looks good to me

offirko reviewed (approved):
Fix for address MSB first looks good

0xc0170 commented:
/morph build

maciejbocianski commented:
tests-mbed_drivers-lp_ticker 
NUCLEO_F746ZG - Toolchain: GCC_ARM
```
[1537279728.87][CONN][RXD] >>> Running case #11: 'Test multi ticker'... 
[1537279728.97][CONN][INF] found KV pair in stream: {{__testcase_start;Test multi ticker}}, queued... 
[1537279729.17][CONN][RXD] :83::FAIL: Expected 16 Was 32 
[1537279729.27][CONN][INF] found KV pair in stream: {{__testcase_finish;Test multi ticker;0;1}}, queued... 
[1537279729.37][CONN][RXD] >>> 'Test multi ticker': 0 passed, 1 failed with reason 'Assertion Failed'
```
I'm looking into it

0xc0170 commented:
@maciejbocianski Let us know if you find anything, we should restart CI here

0xc0170 commented:
/morph test
